src/test/java/io/cdap/plugin/sfmc/sink/DataExtensionClientTest.java

```diff
@@ -0,0 +1,288 @@
+    ETClient client = Mockito.mock(ETClient.class);
+    DataExtensionClient dataExtensionClient = Mockito.spy(new DataExtensionClient(client, dataExtensionKey));
+    PowerMockito.when(ETDataExtension.retrieveColumns(Mockito.any(), Mockito.anyString())).thenReturn(columnList);
+    try {
+      dataExtensionClient.validateSchemaCompatibility(schema, collector);
+      collector.getOrThrowException();
+      Assert.fail("Schema field is null");
+    } catch (ValidationException e) {
+      Assert.assertEquals("Errors were encountered during validation.", e.getMessage());
+    }
+  }
+
+  @Test
+  public void testScan() throws ETSdkException {
+    String dataExtensionKey = "DE";
+    List<ETDataExtensionRow> row = new ArrayList<>();
+    ETDataExtensionRow row1 = new ETDataExtensionRow();
+    row1.setId("id");
+    row.add(row1);
+    ETResponse<ETDataExtensionRow> response = new ETResponse<>();
+    response.setStatus(ETResult.Status.ERROR);
+    PowerMockito.mockStatic(ETDataExtension.class);
+    PowerMockito.when(ETDataExtension.select((ETClient) Mockito.any(), Mockito.anyString())).thenReturn(response);
+    ETClient client = Mockito.mock(ETClient.class);
+    DataExtensionClient dataExtensionClient = Mockito.spy(new DataExtensionClient(client, dataExtensionKey));
+    Assert.assertNotNull(dataExtensionClient.scan());
+  }
+
+  @Test
+  public void testInsert() throws Exception {
+    String requestId = "requestId";
+    String status = "OK";
+    String dataExtensionKey = "DE";
+    List<ETDataExtensionRow> row = new ArrayList<>();
+    ETDataExtensionRow row1 = new ETDataExtensionRow();
+    row1.setId("id");
+    row.add(row1);
+    ETResponse<ETDataExtensionRow> response = new ETResponse<>();
+    response.setStatus(ETResult.Status.ERROR);
+    PowerMockito.mockStatic(ETDataExtension.class);
+    PowerMockito.when(ETDataExtension.select((ETClient) Mockito.any(), Mockito.anyString())).thenReturn(response);
+    ETClient client = Mockito.mock(ETClient.class);
+    Soap soap = Mockito.mock(Soap.class);
+    CreateResponse createResponse = Mockito.mock(CreateResponse.class);
+    ETSoapConnection connection = PowerMockito.mock(ETSoapConnection.class);
+    PowerMockito.whenNew(ETSoapConnection.class).withArguments(Mockito.any(), Mockito.anyString()).
+      thenReturn(connection);
+    Mockito.when(connection.getSoap(Mockito.anyString(), Mockito.anyString())).thenReturn(soap);
+    Mockito.when(soap.create(Mockito.any())).thenReturn(createResponse);
+    Mockito.when(createResponse.getRequestID()).thenReturn(requestId);
+    Mockito.when(createResponse.getOverallStatus()).thenReturn(status);
+    Mockito.when(client.getSoapConnection()).thenReturn(connection);
+    DataExtensionClient dataExtensionClient = Mockito.spy(new DataExtensionClient(client, dataExtensionKey));
+    Assert.assertNotNull(dataExtensionClient.insert(row));
+    Mockito.verify(dataExtensionClient, Mockito.times(1)).insert(row);
```

Comment:
You just called dataExtensionClient.insert() in the previous line, so there's no need for verify that ataExtensionClient.insert() was called! (Similar comment for other tests below.) Instead, you should verify that insert actually inserted the row.

Reply:
Removed the verify call and the working of actual methods has already been covered in DataExtensionRecordWriterTest class as stubs.